tun/tun_linux.go

```diff
@@ -269,21 +269,15 @@ func (tun *NativeTun) setMTU(n int) error {
 
 	defer unix.Close(fd)
 
-	// do ioctl call
-	var ifr [ifReqSize]byte
-	copy(ifr[:], name)
-	*(*uint32)(unsafe.Pointer(&ifr[unix.IFNAMSIZ])) = uint32(n)
-	_, _, errno := unix.Syscall(
-		unix.SYS_IOCTL,
-		uintptr(fd),
-		uintptr(unix.SIOCSIFMTU),
-		uintptr(unsafe.Pointer(&ifr[0])),
-	)
-
-	if errno != 0 {
-		return fmt.Errorf("failed to set MTU of TUN device: %w", errno)
+	req, err := unix.NewIfreq(name)
+	if err != nil {
+		return fmt.Errorf("unix.NewIfreq(%q): %w", name, err)
+	}
+	req.SetUint32(uint32(n))
```

Comment:
technically this is an int field, and i'm not sure that x/sys is correct on systems that have word/byte endian discrepancies, depending on compiler output behaviors

that said, if there is still a bug we should fix it upstream

Reply:
> technically this is an int field

Which is 32 bits on all platforms I know us to target, right? And yes, it's signed, but we never have 2 GB or negative MTUs.

> and i'm not sure that x/sys is correct on systems that have word/byte endian discrepancies

I don't follow. There aren't systems that encode a `uint32` and 32-bit `int` differently in memory, are there? (for the range of values we're talking about)

Reply:
there are two problems, representation, and operation modes. my comment was about operation modes: there are arm modes in which byte operations and word operations produce different bit patterns. the writes through the original code here are byte oriented, but the writes through the x/sys approach are word oriented (so they're actually fine and you can disregard my original comment - the x/sys code should be fine)

Reply:
Ah, gotcha. I follow now.

Meanwhile I was just testing this on Linux and can confirm it works. I even tweaked it to fail & set the MTU+1 to see it go to 1281, etc.

Reply:
Nice, glad to see this API be of use. If it turns out that another representation is more suitable we can always add more setters in x/sys. ifreq is a rough API and this was the best idea I had at making it palatable in Go.

Reply:
@mdlayher just curious, the specifically sized slicing in there, was that just defensive or is there an additional purpose it serves (like hinting something to the compiler or the like)? I read it too fast at the end of a day without eating, and got kind of distracted by it - it's not a problem at all, but now I'm curious?

Reply:
Defensive yep. Trying to be as clear and safe as possible.

Reply:
How'd you even find this PR, @mdlayher? 😅

Reply:
I think because I follow you on github!